### ISSUE TITLE
F-Droid

### ISSUE THREAD

**Issue by linsui** [jakepurple13/OtakuWorld#441]
**Is your feature request related to a problem? Please describe.**
Could you please publish the app on F-Droid? Thanks!

**Describe the solution you'd like**
I can help package and submit it.

**Comment by jakepurple13** [comment]
I would love some help. I've looked into it and couldn't really understand what needed to be done. However, I haven't really been able to work on this project the last couple of days and the next few weeks. I should be primed to continue working by next month.

**Comment by linsui** [comment]
I can help you! I'll package it and make an MR. Thanks!

**Comment by linsui** [comment]
Is that possible to remove Firebase? It's non-free and not allowed in F-Droid.

**Comment by jakepurple13** [comment]
It is...The reason why I added it in the first place is so that people can log in and gain access to their favorites and things read/watched without needing to do steps to transfer data. Plus, crashlytics is a huge plus for when debugging crashes.
What I can try to do is modify the builds so there's a firebase version and a non-firebase version with descriptions of what the difference between the two are.

**Comment by linsui** [comment]
> What I can try to do is modify the builds so there's a firebase version and a non-firebase version with descriptions of what the difference between the two are.

A flavor without firebase/google play service/mlkit would be good. Thanks!

**Comment by jakepurple13** [comment]
I'll see what I can do. It probably won't be easy.

**Comment by linsui** [comment]
Thanks!

**Comment by jakepurple13** [comment]
Somehow I think I was able to do it!
https://github.com/jakepurple13/OtakuWorld/tree/no_firebase_variant
That's the branch right now. I'll merge it either later today or tomorrow. This should also make it easier for others to help work on the app too since the no firebase variant doesn't need firebase.

**Comment by jakepurple13** [comment]
Hm...I'll see what I can do then.

**Comment by linsui** [comment]
Thanks!

**Comment by jakepurple13** [comment]
I think it should be all good now.

**Comment by linsui** [comment]
It works, thanks! It seems there are several different apps in this repo. Is `com.programmersbox.otakuworld` a unified app? Should I package all of them?

**Comment by jakepurple13** [comment]
There's three (technically four) apps in this project. AnimeWorld, AnimeWorld TV, MangaWorld, and NovelWorld. OtakuWorld WILL NOT work. It was an attempt at viewing favorites from all the other apps in one place but it didn't work out too well.

**Comment by linsui** [comment]
Got it! Thanks!

**Comment by linsui** [comment]
AnimeWorld depends on `com.google.android.exoplayer:extension-cast` which depends on GMS. MangaWorld build failed:

```
> Task :mangaworld:processNoFirebaseReleaseResources FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':mangaworld:processNoFirebaseReleaseResources'.
> A failure occurred while executing com.android.build.gradle.internal.res.LinkApplicationAndroidResourcesTask$TaskAction
   > Android resource linking failed
     ERROR:/builds/linsui/fdroiddata/build/com.programmersbox.mangaworld/mangaworld/src/main/res/layout/page_end_chapter_item.xml:9: AAPT: error: attribute adSize (aka com.programmersbox.mangaworld:adSize) not found.
         
     ERROR:/builds/linsui/fdroiddata/build/com.programmersbox.mangaworld/mangaworld/src/main/res/layout/page_end_chapter_item.xml:9: AAPT: error: attribute adUnitId (aka com.programmersbox.mangaworld:adUnitId) not found.
         
     ERROR:/builds/linsui/fdroiddata/build/com.programmersbox.mangaworld/mangaworld/src/main/res/layout/page_next_chapter_item.xml:8: AAPT: error: attribute adSize (aka com.programmersbox.mangaworld:adSize) not found.
         
     ERROR:/builds/linsui/fdroiddata/build/com.programmersbox.mangaworld/mangaworld/src/main/res/layout/page_next_chapter_item.xml:8: AAPT: error: attribute adUnitId (aka com.programmersbox.mangaworld:adUnitId) not found.
         
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org/
BUILD FAILED in 8m 6s
```

Could you please take a look? Thanks!

**Comment by jakepurple13** [comment]
Mmmm...That's annoying. Guess I have to limit the ads too then...Not like they were in the way! If you were wondering how could I do such a thing. They are SO out of the way, you kinda need to try to find them. But I can try something.

As for casting, that's really annoying. That would mean we won't be able to cast. I guess I can, but then need to make sure to explicitly point out that in the no firebase version, casting is not available.

**Comment by linsui** [comment]
Yes, there are many convenient functions provided by GMS and Firebase. Instead of remove all GMS from the flavor, we can also use https://github.com/microg/GmsCore.

**Comment by Efreak** [comment]
You can't get into IzzyOnDroid as they have a maximum of 30mb per app, however you *can* [run an fdroid repo in a github action](https://github.com/xarantolus/fdroid) if you want.

Alternatively, you can use [Obtainium](https://github.com/ImranR98/Obtainium). Try importing this [source file](https://github.com/user-attachments/files/24090168/obtainium-export-2025-12-10T15-55-43.284302-count-3.json) into Obtainium.[^1] This source file filters apks by name for the full releases, which means it will break if you start uploading files with different names.

[^1]: Go to import/export and click obtainium import.
